specs/pvs/corla.pvs

```diff
@@ -2470,6 +2495,181 @@ END data_model
+  asm_state: TYPE
+  asm_event: TYPE
+  asm_transition_function: TYPE =
+    [asm_state, asm_event, boolean -> asm_state]
+%%% Consequently, the core features of the ASM implementation in the
+%%% business logic of the RLA Tool are:
+%%% \begin{itemize}
+%%% \item What is the current state of the system?
+  current_state: TYPE = [dashboard -> asm_state]
+%%% \item What events are enabled?
+  enabled_events: TYPE = [dashboard -> set[asm_event]]
+%%% \item Transition to the next state based upon this event.
+  transition: TYPE = [[dashboard, asm_event] -> [dashboard, asm_state]]
+%%% \item Are we in a initial state?
+  initial?: TYPE = [dashboard -> boolean]
+%%% \item Are we in a final state?
+  final?: TYPE = [dashboard -> boolean]
+%%% \end{itemize}
+%%%
+%%% Also, in order to relate the ASM to the core implementation
+%%% constructs in a RESTful client-server architecture, we must relate
+%%% UI events to ASM events and ASM events to server endpoints.
+  ui_event: TYPE
+  server_endpoint: TYPE
+  ui_to_asm_event_relation: pred[[ui_event, asm_event]]
+  asm_event_to_endpoint_relation: pred[[asm_event, server_endpoint]]
+%%% We will first specify the ASM for the Department of State
+%%% dashboard. Its states describe the sequence of steps that state
+%%% administrators must take to start, observe, and complete an RLA.
+%%% The state \texttt{initial} is the initial state of the ASM; the
+%%% state \texttt{audit\_results\_published} is the final state of the
+%%% ASM.
+  department_of_state_dashboard_state: TYPE = % SUBTYPE asm_state
+    { i_s,    % initial_state
+      a_s,    % authenticated
+      rls_s,  % risk_limits_set
+      ctai_s, % contests_to_audit_identified
+      rsp_s,  % random_seed_published
+      bod_s,  % ballot_order_defined
+      art_s,  % audit_ready_to_start
+      ao_s,   % audit_ongoing
+      ac_s,   % audit_complete
+      arp_s   % audit_results_published
+    }
+%%% Various events, most of which are triggered by state
+%%% administrators using the RLA Tool, cause state changes. These
+%%% events are named according to their corresponding functional
+%%% definitions in earlier theories. E.g., the
+%%% \texttt{authenticate_state_administrator_event} relates to the
+%%% \texttt{authenticate_state_administrator} function and its
+%%% realization via an endpoint.
+  department_of_state_event: TYPE = % SUBTYPE asm_event
+    { asa_e,    % authenticate_state_administrator_event
+      erlfca_e, % establish_risk_limit_for_comparison_audits_event
+      scfca_e,  % select_contests_for_comparison_audit_event
+      ps_e,     % public_seed_event
+      pbta_e,   % publish_ballots_to_audit_event
+      ifhcc_e,  % indicate_full_hand_count_contest
+      r_e       % refresh
+    }
+
+  dos_state: VAR department_of_state_dashboard_state
+  dos_event: VAR department_of_state_event
+  dos_asm_function: VAR asm_transition_function
+
+%   dos_dashboard_next_state(dos_state, dos_event, dos_asm_function):
```

Comment:
Again, this is something that I got told to use dos vs. sos, and now I'm hearing the opposite.  I'm not going to spend a minute worrying about this until stage-3.